test-cases/gutenberg/group.md

```diff
@@ -0,0 +1,355 @@
+
+### Deep nesting is possible
+
+-   Add a `Group` block 
+-   Nest blocks inside multiple `Group` blocks (at least 3 levels deep)
+-   Check if app do not crashes
+
+--------------------------------------------------------------------------------
+
+##### TC008
+
+### Check if Group placeholder is render on RootList
+
+-   Add a `Group` block 
+-   Deselect
+-   Check if you are able to see `Group` placeholder
+
+Expected look:
+ << SCREENSHOT >>
+
+--------------------------------------------------------------------------------
+
+##### TC009
+
+### Check if Group placeholder is render in nested structure
+
+-   Add a `Group` block 
+-   Nest some `Group` inside
+-   Select top-most `Group` block in hierarchy
+-   Check if you are able to see `Group` placeholder of nested blocks wrapped with dashed border
+-   Go down in the hierarchy observing if margins is keept (the whole UI should remain the same)
+-   After selecting last `Group` you should be able to see it's `AppenderButton`
+
+Expected look:
+ << SCREENSHOT >>
+
+--------------------------------------------------------------------------------
+
+##### TC010
+
+### Insertion point (Separator) is show in correct place in RootList
+
+-   Open new post
+-   See if "Start writing" prompt is visible
+-   Press `DefaultAppender`
+-   Check if `Separator` is visible right below the title block
+-   Check if "Start writing" prompt is not visible
+
+<br />
+
+-   Open new post
+-   Type some text in the "Start writing" prompt
+-   Press `DefaultAppender`
+-   Check if `Separator` is visible right below the `Paragraph` block
+
+<br />
+
+-   Add a `Group` block 
+-   Long press `DefaultAppender`
+-   Select where you want to add block after or before
+-   Check if the `Separator` is render outside before/after the `Group`
+
+--------------------------------------------------------------------------------
+
+##### TC011
+
+### Insertion point (Separator) is show in correct place in Group
+
+-   Add a `Group` block 
+-   Add block inside
+-   Select `Group` block
+-   Press group `AppenderButton ( + )`
+-   Check if the `Separator` is render after last block in the `Group`
+
+<br />
+
+-   Add a `Group` block 
+-   Add block inside and keep newly added block selected
+-   Press `AppenderButton ( + )`
+-   Select where you want to add block after or before
+-   Check if the `Separator` is render after/before selected block in the `Group`
+
+--------------------------------------------------------------------------------
+
+##### TC012
+
+### Nested block have proper border styling
+
+-   Add a `Group` block 
+-   Nest some blocks inside
+-   Check if each selected block gets solid blue border
+-   Check if each first child gets dashed grey border
+-   Check if margins and paddings do not bounce when changing selection (exception is the trasition of the selected block connected with making space for FloatingToolbar)
+-   Check if there is no weird/unexpected extra free space around blocks
```

Comment:
It will be difficult to separate margins and padding without  going to debbuger inspector and check exact values and that's not the point here.

We care about if total spaces around block  in selected vs deselected state are constants. That was my intention writing this

Reply:
We don't have to distinguish margins from paddings in the visuals, it is an implementation detail. 
But we should show 2 things:

1. exactly which space we refer to, otherwise it is hard to understand
2. how much space we expect there to be

Reply:
I have update this one